### DISCUSSION TITLE
Integration with spatie/laravel-permission: Log when permission gets assigned to role

### DISCUSSION THREAD

**Discussion by lucacastelnuovo** [spatie/laravel-activitylog#1150] [Q&A]
Hi,

My project uses both spatie/laravel-activitylog and spatie/laravel-permission.
By extending the `Spatie\Permission\Models\Role` to `App\Models\Role`, I can log when a role gets created and when it gets deleted.

Does somebody know how I would log when permission gets "given" to a role using `givePermissionTo()`?

---
**App\Models\Role.php**
```php
<?php

namespace App\Models;

use Spatie\Activitylog\LogOptions;
use Spatie\Activitylog\Traits\LogsActivity;
use Spatie\Permission\Models\Role as SpatieRole;

class Role extends SpatieRole
{
    use LogsActivity;

    public function getActivitylogOptions(): LogOptions
    {
        return LogOptions::defaults()
            ->logOnly(['name'])
            ->logOnlyDirty()
            ->dontSubmitEmptyLogs()
            ->setDescriptionForEvent(fn (string $eventName) => "role.$eventName");
    }
}
```

**Comment by lucacastelnuovo** [comment]
Link to similar question: https://github.com/spatie/laravel-activitylog/discussions/996

**Reply by onovaes** [reply]
Were you able to log using givePermissionTo()? Help-me

**Reply by lucacastelnuovo** [reply]
@onovaes No I was not :). I would like to implement this feature but did not manage to.
